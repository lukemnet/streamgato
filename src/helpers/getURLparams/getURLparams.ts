const getURLparams = () => {
  const queryString = window.location.search.substring(1);
  const parsedParams = decodeURI(queryString)
    .replace(/"/g, '\\"')
    .replace(/&/g, '","')
    .replace(/=/g,'":"');

  return queryString
    ? JSON.parse(`{"${parsedParams}"}`)
    : {};
};

export default getURLparams;
