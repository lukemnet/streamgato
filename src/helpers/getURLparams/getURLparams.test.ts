import getURLparams from './getURLparams';

it('should create correct param object', () => {
  delete window.location;
  (window.location as any) = { search: "?pt=5&pl=5&pb=10" };

  expect(getURLparams()).toMatchSnapshot();
});

it('should return empty param object when no search params are present', () => {
  delete window.location;
  (window.location as any) = { search: "" };

  expect(getURLparams()).toMatchSnapshot();
})