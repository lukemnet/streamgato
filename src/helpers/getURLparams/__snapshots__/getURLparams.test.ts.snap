// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create correct param object 1`] = `
Object {
  "pb": "10",
  "pl": "5",
  "pt": "5",
}
`;

exports[`should return empty param object when no search params are present 1`] = `Object {}`;
